src/front/glsl/functions.rs

```diff
@@ -195,14 +195,39 @@ impl Parser {
                 (value, expr_meta),
                 meta,
             )?,
-            TypeInner::Struct { .. } | TypeInner::Array { .. } => ctx.add_expression(
-                Expression::Compose {
-                    ty,
-                    components: vec![value],
-                },
-                meta,
-                body,
-            ),
+            TypeInner::Struct { ref members, .. } => {
+                let scalar_components = members
+                    .get(0)
+                    .and_then(|member| scalar_components(&self.module.types[member.ty].inner));
+                if let Some((kind, width)) = scalar_components {
+                    ctx.implicit_conversion(self, &mut value, expr_meta, kind, width)?;
```

Comment:
aren't you changing the second parameter to be `&value` now?

Reply:
No, the update signature is
```diff
pub fn implicit_conversion(
        &mut self,
-       parser: &mut Parser,
+       parser: &Parser,
        expr: &mut Handle<Expression>,
        meta: Span,
        kind: ScalarKind,
        width: crate::Bytes,
    )
```

So only the mutability of the first argument is changed (`self` in this case), while the second argument (`&mut value`) is still a mutable reference.

Otherwise the code wouldn't compile because `implicit_conversion` mutates `expr`